src/dialect/dialect-adapter.ts

```diff
@@ -31,6 +31,12 @@ export interface DialectAdapter {
    */
   readonly supportsReturning: boolean
 
+  /**
+   * Whether or not this dialect supports the `output` clause in inserts
+   * updates and deletes.
+   */
+  readonly supportsOutput: boolean
```

Comment:
Hmm.. I wonder if every 3rd party dialect correctly uses the `DialectAdapterBase`. If not, this is kind of a breaking change. Maybe this should be optional?

Reply:
Good idea! it is optional now.